homeassistant/components/switch/nissan_leaf.py

```diff
@@ -0,0 +1,108 @@
+"""
+Charge and Climate Control Support for the Nissan Leaf.
+
+Please refer to the main platform component for configuration details
+"""
+
+import logging
+from homeassistant.components.nissan_leaf import (
+    DATA_CHARGING, DATA_CLIMATE, DATA_LEAF, LeafEntity
+)
+from homeassistant.helpers.entity import ToggleEntity
+
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['nissan_leaf']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Nissan Leaf switch platform setup."""
+    _LOGGER.debug("In switch setup platform, discovery_info=%s",
+                  discovery_info)
+
+    devices = []
+    for key, value in hass.data[DATA_LEAF].items():
+        _LOGGER.debug("Adding switch for item key=%s, value=%s", key, value)
+        _LOGGER.debug("Adding LeafChargeSwitch(%s)", value)
+        devices.append(LeafChargeSwitch(value))
+        _LOGGER.debug("Adding LeafClimateSwitch(%s)", value)
+        devices.append(LeafClimateSwitch(value))
+
+    _LOGGER.debug("Calling add_devices for switches")
+    add_devices(devices, True)
+
+
+class LeafClimateSwitch(LeafEntity, ToggleEntity):
+    """Nissan Leaf Climate Control switch."""
+
+    @property
+    def name(self):
+        """Switch name."""
+        return self.car.leaf.nickname + " Climate Control"
+
+    def log_registration(self):
+        """Log registration."""
+        _LOGGER.debug(
+            "Registered LeafClimateSwitch component with HASS for VIN %s",
+            self.car.leaf.vin)
+
+    @property
+    def is_on(self):
+        """Return true if climate control is on."""
+        return self.car.data[DATA_CLIMATE] is True
+
+    async def async_turn_on(self, **kwargs):
+        """Turn on climate control."""
+        if await self.car.async_set_climate(True):
+            self.car.data[DATA_CLIMATE] = True
+
+    async def async_turn_off(self, **kwargs):
+        """Turn off climate control."""
+        if await self.car.async_set_climate(False):
+            self.car.data[DATA_CLIMATE] = False
+
+    @property
+    def icon(self):
+        """Climate control icon."""
+        if self.car.data[DATA_CLIMATE]:
+            return 'mdi:fan'
+        return 'mdi:fan-off'
+
+
+class LeafChargeSwitch(LeafEntity, ToggleEntity):
+    """Nissan Leaf Charging On switch."""
+
+    @property
+    def name(self):
+        """Switch name."""
+        return self.car.leaf.nickname + " Charging Status"
+
+    def log_registration(self):
+        """Log registration."""
+        _LOGGER.debug(
+            "Registered LeafChargeSwitch component with HASS for VIN %s",
+            self.car.leaf.vin)
+
+    @property
+    def icon(self):
+        """Charging switch icon."""
+        if self.car.data[DATA_CHARGING]:
+            return 'mdi:flash'
+        return 'mdi:flash-off'
```

Comment:
I'm not sure this is allowed (dynamic change of icons)

Reply:
There's a couple of examples of changing the Icon here:  https://github.com/home-assistant/home-assistant/blob/91aa874c0cc8b4e0daf5a268bc2d3ac4c2339fc2/homeassistant/components/sensor/dsmr.py#L254  and https://github.com/home-assistant/home-assistant/blob/cb07ea0d60acec43778687282504fdf8eb712885/homeassistant/components/switch/android_ip_webcam.py#L90 

Perhaps the `mdi:flash` and `mdi:flash` strings should be constants.